packages/amazonq/src/extensionShared.ts

```diff
@@ -140,21 +142,72 @@ export async function activateShared(context: vscode.ExtensionContext, isWeb: bo
             telemetry.record({ source: ExtStartUpSources.reload })
         }
 
-        const authState = (await AuthUtil.instance.getChatAuthState()).codewhispererChat
         const authKinds: AuthUtils.AuthSimpleId[] = []
         if (await AuthUtils.hasBuilderId('codewhisperer')) {
             authKinds.push('builderIdCodeWhisperer')
         }
         if (await AuthUtils.hasSso('codewhisperer')) {
             authKinds.push('identityCenterCodeWhisperer')
         }
+
+        const authStatus = await getAuthStatus()
+
         telemetry.record({
-            authStatus: authState === 'connected' || authState === 'expired' ? authState : 'notConnected',
+            authStatus,
             enabledAuthConnections: authKinds.join(','),
         })
     })
 }
 
+/** Determines the status of the Auth for Amazon Q that is used at the extension startup. */
+async function getAuthStatus(): Promise<AuthStatus> {
+    const authState = (await AuthUtil.instance.getChatAuthState()).codewhispererChat
+    let authStatus: AuthStatus = authState === 'connected' || authState === 'expired' ? authState : 'notConnected'
+
+    // The Q extension does not have its own extension, so as a fallback we want to check
+    // if the Toolkit extension can provide a connection that works with Q since we will
+    if (authStatus === 'notConnected') {
+        let autoConnectConn: AwsConnection | undefined = undefined
+        try {
+            autoConnectConn = await getAutoConnectableConnection()
+        } catch (e) {
+            getLogger().error(`Failed ${getAutoConnectableConnection.name}:\n\n%s`, JSON.stringify(e))
+        }
+
+        if (autoConnectConn) {
+            authStatus = autoConnectConn.state === 'valid' ? 'connected' : 'expired'
+        }
+    }
+
+    return authStatus
+}
+
+/**
+ * Returns a connection from the standalone toolkit extension that
+ * the Amazon Q extension can use. Otherwise it returns undefined.
+ *
+ * HACK: Our frontend Login ui for Amazon Q will auto connect if required/possible,
+ * but we cannot detect this at the end of Amazon Q extension activation.
+ * So we reuse the same {@link findUsableQConnection}()
+ * and assume that the frontend will have the same result and auto connect.
+ */
+async function getAutoConnectableConnection(): Promise<AwsConnection | undefined> {
```

Comment:
should this and getAuthStatus live in a new `packages/amazonq/src/auth/util.ts` file?